File: 2387-partition-array-such-that-maximum-difference-is-k/partition-array-such-that-maximum-difference-is-k.rb
# @param {Integer[]} nums
# @param {Integer} k
# @return {Integer}
def partition_array(nums, k)
   nums.sort!
    result = 0
    i = 0

    while i < nums.length
        start = nums[i]
        result += 1

        while i < nums.length && nums[i] - start <= k
            i += 1
        end
    end

    result
end

File: 2387-partition-array-such-that-maximum-difference-is-k/partition-array-such-that-maximum-difference-is-k.go
func partitionArray(nums []int, k int) int {
	sort.Ints(nums)
	result, i := 0, 0

	for i < len(nums) {
		start := nums[i]
		result++

		for i < len(nums) && nums[i]-start <= k {
			i++
		}
	}

	return result
}